---
identifier: "1127"
closedAt: null
mergedAt: null
title: "When `sourceStream` errors, yield a `{ errors: [...] }` response"
stage: "0"
champion: benjie
prUrl: https://github.com/graphql/graphql-spec/pull/1127
events:
  - type: wgAgenda
    href: https://github.com/graphql/graphql-wg/blob/main/agendas/2024/12-Dec/05-wg-primary.md
    date: 2024-12-05
    actor: null
  - type: prCreated
    date: 2024-11-21T15:11:21Z
    href: https://github.com/graphql/graphql-spec/pull/1127
    actor: benjie
  - type: commitsPushed
    date: 2024-11-21
    href: https://github.com/graphql/graphql-spec/commit/2ef903c4a4fd6a0793c0274c67a42d53ca419170
    actor: benjie
    commits:
      - href: https://github.com/graphql/graphql-spec/commit/2ef903c4a4fd6a0793c0274c67a42d53ca419170
        headline: Don't complete with error.
        ghUser: benjie
        authorName: Benjie Gillam
      - href: https://github.com/graphql/graphql-spec/commit/5257c92869451ae79134ecd8e3f3b13a6005b652
        headline: Don't use a new algorithm, no need to DRY
        ghUser: benjie
        authorName: Benjie Gillam
      - href: https://github.com/graphql/graphql-spec/commit/f49aab72ea487dd47fc1e1f72dcf939b3d0b221e
        headline: Internal errors should be passed up the chain
        ghUser: benjie
        authorName: Benjie Gillam
related: "1099"
shortname: "When `sourceStream` errors, yield a `{ errors: [...] }` response"
image: /img/rfc_tracker.png
---

## At a glance

- **Identifier**: #1127
- **Stage**: [RFC0: Strawman](https://github.com/graphql/graphql-spec/blob/main/CONTRIBUTING.md#stage-0-strawman)
- **Champion**: [@benjie](https://github.com/benjie)
- **PR**: [When &#x60;sourceStream&#x60; errors, yield a &#x60;&#x7b; errors: &#x5b;...&#x5d; &#x7d;&#x60; response](https://github.com/graphql/graphql-spec/pull/1127)

<!-- BEGIN_CUSTOM_TEXT -->



<!-- END_CUSTOM_TEXT -->

## Timeline

- **Added to [2024-12-05 WG agenda](https://github.com/graphql/graphql-wg/blob/main/agendas/2024/12-Dec/05-wg-primary.md)**
- **[Spec PR](https://github.com/graphql/graphql-spec/pull/1127) created** on 2024-11-21 by benjie
- **3 commits pushed** on 2024-11-21:
  - [Don't complete with error.](https://github.com/graphql/graphql-spec/commit/2ef903c4a4fd6a0793c0274c67a42d53ca419170) by [@benjie](https://github.com/benjie)
  - [Don't use a new algorithm, no need to DRY](https://github.com/graphql/graphql-spec/commit/5257c92869451ae79134ecd8e3f3b13a6005b652) by [@benjie](https://github.com/benjie)
  - [Internal errors should be passed up the chain](https://github.com/graphql/graphql-spec/commit/f49aab72ea487dd47fc1e1f72dcf939b3d0b221e) by [@benjie](https://github.com/benjie)

<!-- VERBATIM -->

---

> See https://github.com/graphql/graphql-spec/pull/1099/files#r1799509253
> 
> Currently if `sourceStream` generates an error, then `responseStream` repeats the error. This is the behavior implemented in graphql-js and [is problematic](https://github.com/graphql/graphql-js/issues/4001).
> 
> GraphQL captures execution errors and wraps them in an `{ errors: [...] }` payload for query and mutation operations; it should do the same for stream errors in a subscription operation.
> 
> This PR makes this change. It is technically a breaking change, I think, so will requireâ€‹ careful thought.
